.github/workflows/go.yml

```diff
@@ -7,7 +7,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        go: [ 1.17.x, 1.18.x ]
+        go: [ 1.18.x, 1.19.x ]
```

Comment:
unrelated nit: Given the sheer amount of highly concurrent code in shared client, we should also run with `go test -race` in this workflow